packages/malloy/src/lang/parse-tree-walkers/find-table-path-walker.ts

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining
+ * a copy of this software and associated documentation files
+ * (the "Software"), to deal in the Software without restriction,
+ * including without limitation the rights to use, copy, modify, merge,
+ * publish, distribute, sublicense, and/or sell copies of the Software,
+ * and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+ * CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+import {MalloyParserListener} from '../lib/Malloy/MalloyParserListener';
+import {
+  HasString,
+  getId,
+  getStringIfShort,
+  getStringParts,
+  unIndent,
+} from '../parse-utils';
+import {MalloyTranslation} from '../parse-malloy';
+import {CommonTokenStream} from 'antlr4ts';
+import {DocumentRange} from '../../model/malloy_types';
+import * as parser from '../lib/Malloy/MalloyParser';
+import {MalloyParseInfo} from '../malloy-parse-info';
+import {ParseTreeWalker} from 'antlr4ts/tree/ParseTreeWalker';
+
+export interface PathInfo {
+  connectionId: string;
+  tablePath: string;
+  range: DocumentRange;
+}
+
+class FindTablePathWalker implements MalloyParserListener {
+  pathInfos: PathInfo[] = [];
+  constructor(
+    readonly translator: MalloyTranslation,
+    readonly tokens: CommonTokenStream
+  ) {}
+
+  protected getPlainString(cx: HasString): string | undefined {
```

Comment:
It might be nice to abstract this out into a utility function in `parse-utils`, since `malloy-to-ast` has almost exactly the same function.

Reply:
Done